File: 30 Days of Code/PHP/Day 19 - Interfaces.php
<?php
interface AdvancedArithmetic{
    public function divisorSum($n);
}

/*
 * Write your code here
 */
class Calculator implements AdvancedArithmetic {
    function divisorSum($n) {
        if ($n==1)
            return 1;
        $i = 1;
        $s = 0;
        while ($i<=(int)sqrt($n)){
            if ($n%$i==0){
                if ($n/$i == $i)
                    $s = $s + $i;
                else
                    $s = $s + ($n/$i) + $i;}
            $i += 1;
        }
        return $s;
    }
}

$n=intval(fgets(STDIN));
$myCalculator=new Calculator();
if($myCalculator instanceof AdvancedArithmetic)//checking if Calculator has implemented AdvancedArithemtic
{
    $sum=$myCalculator->divisorSum($n);
    echo "I implemented: AdvancedArithmetic\n".$sum;
}
else
{
    echo "Wrong answer";// You will get this output if you dont implement
}
?>

File: 30 Days of Code/Python/Day 19 - Interfaces.py
class AdvancedArithmetic(object):
    def divisorSum(n):
        raise NotImplementedError

class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        if n==1:
            return 1
        i = 1
        s = 0
        while i<=int(n**(0.5)):
            if n%i==0:
                if int(n/i) == i:
                    s = s + i
                else:
                    s = s + int(n/i) + i
                #print(i,int(n/i))
            i += 1
        return s
            


n = int(input())
my_calculator = Calculator()
s = my_calculator.divisorSum(n)
print("I implemented: " + type(my_calculator).__bases__[0].__name__)
print(s)